src/components/TodoItem.tsx

```diff
@@ -0,0 +1,116 @@
+import cn from 'classnames';
+
+import { Todo } from '../types/Todo';
+import { ChangeEvent, FormEvent, useState } from 'react';
+
+type Props = {
+  todo: Todo;
+  setSelectedTodo: (todo: Todo) => void;
+  setUpdatingTodo: (updatingTodo: number | null) => void;
+  updatingTodo: number | null;
+  isDeleting?: boolean;
+  selectedTodo: Todo | null;
+  onDelete: (id: number) => void;
+  onPatch: (
+    event: React.FormEvent<HTMLFormElement> | ChangeEvent<HTMLInputElement>,
+    todo: Todo,
+  ) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  setSelectedTodo,
+  setUpdatingTodo,
+  updatingTodo,
+  selectedTodo,
+  onDelete,
+  onPatch,
+}) => {
+  const [isDoubleClicked, setIsDoubleClicked] = useState(false);
+
+  const onSelect = (todoItem: Todo) => {
+    setUpdatingTodo(todoItem.id);
+    setSelectedTodo(todoItem);
+  };
+
+  const onFormSubmit = (event: FormEvent<HTMLFormElement>) => {
+    if (selectedTodo) {
+      setIsDoubleClicked(false);
+      onPatch(event, selectedTodo);
+    }
+  };
+
+  const isUpdatingTodo = !updatingTodo || updatingTodo !== todo.id;
+  const isSelectedTodo = !selectedTodo || selectedTodo.id !== todo.id;
+
+  return (
+    <div data-cy="Todo" className={cn('todo', { completed: todo.completed })}>
+      <label className="todo__status-label">
```

Comment:
A form label must be associated with a control
![image](https://github.com/mate-academy/react_todo-app-loading-todos/assets/63243136/a3de6a18-f46e-4e5a-8c24-928d8907d14b)